### PR TITLE
Added Kotlin Cheat Sheet

### PR DESCRIPTION
**Pull request by AdamMc331**
Includes classes/functions/variables/control flow. Would love to add a section in the future about destructuring, delegates, and some other helpful tricks but wanted to get out a starting point.

### CODE REVIEW

File: kotlin.md [hunk 1]
```diff
@@ -0,0 +1,385 @@
...
+### Filtering & Searching
+
+```kotlin
+val evenNumbers = numList.filter { it % 2 == 0 }
+val containsEven = numList.any { it % 2 == 0 }
+val containsNoEvens = numList.none { it % 2 == 0 }
+val containsNoEvens = numList.all { it % 2 == 1 }
+val firstEvenNumber: Int = numList.first { it % 2 == 0 }
+val firstEvenOrNull: Int? = numList.firstOrNull { it % 2 == 0 }
+```
```
Comment (MclaughlinSteve) [inline kotlin.md]: Consider adding a small section just demonstrating what `it` is

**Comment by AdamMc331** [comment]
Rebased and squashed my commits for this PR.. I don't plan on making any more immediate changes unless they're requested by someone. :) 

@rstacruz please let me know if there's anything else I need to add to this!

File: kotlin.md [hunk 2]
```diff
@@ -0,0 +1,400 @@
...
+class Person(val name: String) {
+    private var age: Int? = null
+
+    constructor(name: String, age: Int) : this(name) {
+        this.age = age
+    }
+}
+
+// Above can be replaced with default params
+class Person(val name: String, val age: Int? = null)
```
Comment (asad-awadia) [inline kotlin.md]: Would be nice to have a section on init

**Comment by AdamMc331** [comment]
@rstacruz Bumping this one. Let me know what you think. :)

**Comment by chadwithuhc** [comment]
Thanks @AdamMc331 !!

**Pull request merged**